File: cache/src/main/java/com/mm/http/cache/CacheStrategy.java
package com.mm.http.cache;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;
import java.util.concurrent.TimeUnit;

/**
 * Created by : majian
 * Date : 2021/8/24
 */

@Target({ElementType.TYPE, ElementType.METHOD})
@Retention(RUNTIME)
public @interface CacheStrategy {
    @StrategyType
    int value() default StrategyType.NO_CACHE;

    long duration() default 24L;

    TimeUnit timeUnit() default TimeUnit.HOURS;
}

File: cache/src/main/java/com/mm/http/cache/CacheStrategy.kt
package com.mm.http.cache

import java.util.concurrent.TimeUnit

/**
 * Created by : majian
 * Date : 2021/8/24
 */
@Target(
    AnnotationTarget.ANNOTATION_CLASS,
    AnnotationTarget.CLASS,
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER
)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class CacheStrategy(
    val value: Int = StrategyType.NO_CACHE,
    val duration: Long = 24L,
    val timeUnit: TimeUnit = TimeUnit.HOURS
)